gui/web/src/utils/metricsTracker.js

```diff
@@ -0,0 +1,27 @@
+class MetricsTracker {
+    constructor(apiKey) {
+        this.apiKey = apiKey;
+        this.apiUrl = "https://api2.amplitude.com/2/httpapi";
+
+        // TODO: Implement more functions.
+        this.sendEvent = this.sendEvent.bind(this);
+    }
+
+    sendEvent(event) {
+        var _this = this
+        this._asyncRequests["fetch"] = fetch(this.apiUrl, {
+            method: "POST",
+            eventName: event,
+        }).then(resp => {
+            if (!_this._asyncRequests["fetch"]) {
+                // if it has been deleted, we don't want to process the result
+                return
+            }
+            delete _this._asyncRequests["fetch"];
+
+            return resp.json();
```

Comment:
we don't want to return the response really for sending events, unless there's an error. In the case where there's an error we can log to the console (because these are metric errors and should not be shown to the user in the GUI). In all other cases when it succeeds we can just ignore it really.